### DISCUSSION TITLE
Why doesn't `StateNotifier` have an internal `Ref` object, by default?

### DISCUSSION THREAD

**Discussion by lucavenir** [rrousselGit/riverpod#1407] [Ideas]
Hi there,

I'm just curious: why do we rely on passing around the `ref` object, whereas we could have it in `StateNotifier` by default? Is it possible, does this make sense?

Using it in a `StateNotifier` is kind of a common practice (from my understanding), so I'm just asking this out of curiosity.

**Comment by rrousselGit** [comment]
StateNotifier is from a separate package, and therefore doesn't know about Ref

**Reply by lucavenir** [reply]
Oh. Sorry, I had no idea. Thank you!
